mrpt_tps_astar_planner/src/mrpt_tps_astar_planner_node.cpp

```diff
@@ -607,21 +609,13 @@ void TPS_Astar_Planner_Node::update_obstacles(
 	// Transform the cloud to its global pose in the map:
 	mrpt::poses::CPose3D sensorPoseInMap;
 
-	// Brief pause to allow time for the transform data to become available
-	const auto timeout = std::chrono::milliseconds(50);
-	const auto tStart = this->now();
-	const double max_duration = 5.0;  // seconds
-
-	while (!wait_for_transform(sensorPoseInMap, pcMsg->header.frame_id, frame_id_map_))
+	// Update fields only when transform data to become available
+	if (wait_for_transform(sensorPoseInMap, pcMsg->header.frame_id, frame_id_map_))
 	{
-		std::this_thread::sleep_for(timeout);
-		auto duration = this->get_clock()->now() - tStart;
-		ASSERT_(duration.seconds() < max_duration);
-	}
+		pc->changeCoordinatesReference(sensorPoseInMap);
 
-	pc->changeCoordinatesReference(sensorPoseInMap);
-
-	e.obstacle_points = pc;
+		e.obstacle_points = pc;
```

Comment:
Maybe in the "else" we could clear the existing point cloud, just to make sure it's not used as if the robot was at a wrong past location? Not needed for this PR, but think about that possibility...